dashboard/src/reducers/apps.ts

```diff
@@ -30,6 +31,22 @@ const appsReducer = (
       return { ...state, isFetching: true, listingAll: action.payload };
     case getType(actions.apps.receiveAppList):
       return { ...state, isFetching: false, listOverview: action.payload };
+    case getType(actions.apps.receiveAppUpdateInfo):
+      const stateCopy = cloneDeep(state);
```

Comment:
Deep cloning is expensive, and seems pretty unnecessary here __just__ to update one app's `updateInfo` field. See https://medium.freecodecamp.org/handling-state-in-react-four-immutable-approaches-to-consider-d1f5c00249d5. I would try to use the previously suggested approaches (https://gist.github.com/gorangajic/e902c2ee994260b3348d#update-array-with-objects-using-arrayprototypeslice), but we could discuss whether it's worth adding an immutability helper.

@Angelmmiguel @earnubs any thoughts on this?

Reply:
Just did a bit more digging on this, Redux also has some excellent documentation on different approaches to dealing with this problem: https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns#inserting-and-removing-items-in-arrays. For example, one option for the `listOverview` is use `Array.map` to update the one we want: https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns#updating-an-item-in-an-array.

Another way to help make this a little clearer would be to add a new reducer that just acts on the `listOverview`, and perhaps another one that acts on `selected`.

Reply:
okay, honestly I don't think that using `cloneDeep` would create a bottleneck but it's okay for me to sacrifice the readability/simplicity.

Note, due to the type of `state.selected` I am not able to do:

```
selected = {...state.selected, updateInfo};
```

It fails with:

```
Type 'IRelease | { updateInfo: IChartUpdateInfo; name: string; info?: IInfo | null | undefined; chart?: IChart | null | undefined; config?: IConfig | null | undefined; manifest: string; hooks: IHook[]; version: number; namespace: string; } | undefined' is not assignable to type 'IRelease | undefined'.
  Property 'toJSON' is missing in type '{ updateInfo: IChartUpdateInfo; name: string; info?: IInfo | null | undefined; chart?: IChart | null | undefined; config?: IConfig | null | undefined; manifest: string; hooks: IHook[]; version: number; namespace: string; }' but required in type 'IRelease'.
```

I guess the operator `...` does not play good with methods of a class

Reply:
Perhaps not now or in your testing, but what happens on a cluster where there are a lot of apps? What happens when we later expand AppOverview to hold more data. It seems like a bad practice and the alternatives are not necessarily that much worse.

Reply:
Yes, I know, thanks for the suggestion!